File: module_tophome/src/main/java/com/wavvy/tophome/NetService.java
package com.wavvy.tophome;
import retrofit2.http.GET;
/*
 * @Module     : com.wavvy.tophome
 * @FileName   : NetService
 * @Author     : lsj
 * @CreateTime : 2026/7/16 10:45
 * @Desc       :
 */
interface NetService {
    @GET("playlist/hot")
}

File: module_tophome/src/main/java/com/wavvy/tophome/NetService.kt
package com.wavvy.tophome

import com.wavvy.tophome.data.BannerData
import com.wavvy.tophome.data.DetailData
import com.wavvy.tophome.data.LoveSongData
import com.wavvy.tophome.data.SongSeaData
import retrofit2.http.GET
import retrofit2.http.Query

/*
 * @Module     : com.wavvy.tophome
 * @FileName   : NetService
 * @Author     : lsj
 * @CreateTime : 2026/7/16 14:58
 * @Desc       : 
 */
interface NetService {
    @GET("toplist/detail")
    suspend fun getDetails() : DetailData
    @GET("banner")
    suspend fun getBannerList(@Query("type") type: Int = 1): BannerData
    @GET("recommend/songs")
    suspend fun getRecommendSongs() : SongSeaData
    @GET("personalized")
    suspend fun getLoveSong(): LoveSongData
}
